src/scriptsys/scr_cvar.c

```diff
@@ -0,0 +1,161 @@
+#include "lua/lua.h"
+#include "lua/lualib.h"
+#include "lua/lauxlib.h"
+#include "math/types.h"
+#include "common/console.h"
+#include "common/cvar.h"
+#include "scr_cmd.h"
+#include "script.h"
+
+static void scr_push_vec(lua_State* L, float4 vec)
+{
+	lua_createtable(L, 4, 0);
```

Comment:
table == {} or an object in lua, right?
A TODO for later: we should figure out a faster way to pass structs to lua, maybe via FFI, or maybe just push 4 floats?
You may want to use VEC_CALL on anything taking or returning a math type like float4 (unless its passed as a function pointer to lua).

Reply:
I think we could use `UserData` for this, and then have the X, Y, Z, W be fetched using metatables. Would have to profile performance!

Reply:
#23